src/components/badge.tsx

```diff
@@ -0,0 +1,37 @@
+import {ReactNode} from 'react';
+import {XMarkIcon} from '@heroicons/react/24/solid';
+
+interface Props {
+  children: ReactNode;
+  'data-test'?: string;
+}
+
+export default function Badge({children, 'data-test': dataTest}: Props) {
+  return (
+    <span
+      data-test={dataTest}
+      className="m-1 inline-flex items-center rounded-full border border-gray-200 bg-white py-1.5 pl-3 pr-2 text-xs font-medium text-gray-900"
+    >
+      {children}
+    </span>
+  );
+}
+
+interface TagActionProps {
+  Icon?: React.ElementType;
+  onClick: () => void;
+}
+
+function BadgeAction({Icon = XMarkIcon, onClick}: TagActionProps) {
+  return (
+    <button
+      type="button"
+      className="ml-1 inline-flex flex-shrink-0 rounded-full p-1 text-gray-400 hover:bg-gray-200 hover:text-gray-500"
+      onClick={onClick}
+    >
+      <Icon className="h-3 w-3" />
+    </button>
+  );
+}
+
+Badge.Action = BadgeAction;
```

Comment:
For my understanding: is this a way to export two components while using one default export? If so, is this preferred to using named exports [1]?

[1] https://beta.reactjs.org/learn/importing-and-exporting-components#default-vs-named-exports

Reply:
I like writing global components this way. In my previous team, we called this the [Headless UI](https://headlessui.com/react/menu) pattern. Not sure what the official name is.

First, separate the individual parts into little components. So when you use the global component you can choose which parts you need, and choose the order of the individual parts.

Because of this, you can have a lot of little components. To communicate that the individual components do belong to each other choose one name for the parent component. In this case `Badge`. And name the child components `Badge.ChildName`.

So it's mostly to communicate that the Child should only be used as part of the parent.

Headless UI uses this pattern to add logic to the wrapper and child components, but it also works great for UI components.

Reply:
Here is some documentation. I will review these and see which is best to put in the documentation.

- https://reactjs.org/docs/jsx-in-depth.html#using-dot-notation-for-jsx-type
- https://blog.logrocket.com/react-design-patterns/#compound-components-pattern
- https://andreidobrinski.com/blog/react-component-composition-with-dot-notation-exports/
- https://dev.to/alexandprivate/react-dot-notation-component-with-ts-49k8
- https://medium.com/risan/react-component-with-dot-notation-7a9853dbf33b

Reply:
Moved the writing of the documentation to: https://github.com/colonial-heritage/dataset-browser/issues/93